core/spec/lib/spree/core/testing_support/factories/order_factory_spec.rb

```diff
@@ -1,6 +1,98 @@
 require 'spec_helper'
 require 'spree/testing_support/factories/order_factory'
 
+RSpec.shared_examples "an order with line items factory" do |expected_order_state, expected_inventory_unit_state|
+  # This factory cannot be built correctly because Shipment#set_up_inventory
+  # requires records to be saved.
+  context "when created" do
+    let(:stock_location) { create(:stock_location) }
+    let(:first_variant) { create(:variant) }
+    let(:second_variant) { create(:variant) }
+    let(:shipping_method) { create(:shipping_method) }
+    let(:order) do
+      create(
+        factory,
+        stock_location: stock_location,
+        line_items_attributes: [
+          { variant: first_variant, quantity: 1, price: 1 },
+          { variant: second_variant, quantity: 2, price: 2 }
+        ],
+        shipment_cost: 3,
+        shipping_method: shipping_method
+      )
+    end
+
+    it "has the expected attributes" do
+      aggregate_failures "for line items" do
+        expect(order.line_items.count).to eq 2
+        expect(order.line_items[0]).to have_attributes(
+          quantity: 1,
+          price: 1.0
+        )
+        expect(order.line_items[1]).to have_attributes(
+          price: 2.0,
+          quantity: 2
+        )
+      end
+
+      aggregate_failures "for shipments" do
+        expect(order.shipments.count).to eq 1
+        expect(order.shipments[0]).to have_attributes(
+          amount: 3.0,
+          stock_location: stock_location
+        )
+
+        expect(order.shipments[0].shipping_method).to eq(shipping_method)
+
+        expect(order.shipments[0].inventory_units.count).to eq(3)
+        expect(order.shipments[0].inventory_units[0]).to have_attributes(
+          order: order,
+          shipment: order.shipments[0],
+          line_item: order.line_items[0],
+          variant: order.line_items[0].variant,
+          state: expected_inventory_unit_state
+        )
+        expect(order.shipments[0].inventory_units[1]).to have_attributes(
+          order: order,
+          shipment: order.shipments[0],
+          line_item: order.line_items[1],
+          variant: order.line_items[1].variant,
+          state: expected_inventory_unit_state
+        )
+        expect(order.shipments[0].inventory_units[2]).to have_attributes(
+          order: order,
+          shipment: order.shipments[0],
+          line_item: order.line_items[1],
+          variant: order.line_items[1].variant,
+          state: expected_inventory_unit_state
+        )
+      end
+
+      expect(order).to have_attributes(
+        item_total: 5.0,
+        ship_total: 3.0,
+        total: 8.0,
+        state: expected_order_state
+      )
+    end
+  end
+
+  context 'when shipments should be taxed' do
+    let!(:ship_address) { create(:address) }
+    let!(:tax_zone) { create(:global_zone) } # will include the above address
+    let!(:tax_rate) { create(:tax_rate, amount: 0.10, zone: tax_zone, tax_categories: [tax_category]) }
```

Comment:
I'm trying to backport this PR to Solidus 1.3 to help with an upgrade I'm working on. I needed to change `tax_categories` here to `tax_category`, but it seems taxes aren't being added to shipments in my backport. Any thoughts?